Error: Cannot find module 'prettier' with @wordpress/eslint-plugin v3.4.0

**Describe the bug**

See this example repository with the minimum setup to replicate the issue:
https://github.com/kasparsd/wordpress-eslint-plugin-prettier

Running `eslint` with `.estlintrc.js` config set to:

```json
{
	"extends": [ "plugin:@wordpress/eslint-plugin/recommended" ]
}
```

and `@wordpress/eslint-plugin` as a development dependency set to version `^3.4.0` produces the following error:

```
Error: Cannot find module 'prettier'
Occurred while linting /Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/src/script.js:1
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Program (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js:162:26)
    at listeners.(anonymous function).forEach.listener (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/node-event-generator.js:283:22)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @kasparsd/wp-scripts-eslint-prettier-bug@1.0.0 lint: `eslint ./src`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @kasparsd/wp-scripts-eslint-prettier-bug@1.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kaspars/.npm/_logs/2020-02-04T10_54_35_609Z-debug.log
```

See the output on Travis:
https://travis-ci.com/kasparsd/wp-scripts-eslint-prettier-bug/jobs/283358536#L519-L538

Downgrading `@wordpress/eslint-plugin` to version `3.3.0` fixes the problem -- see this pull request:

https://github.com/kasparsd/wordpress-eslint-plugin-prettier/pull/1

and the Travis CI output for the change:

https://travis-ci.com/kasparsd/wp-scripts-eslint-prettier-bug/jobs/283361201#L463-L467


**To reproduce**

Steps to reproduce the behavior:

1. Clone the sample repository `git clone https://github.com/kasparsd/wordpress-eslint-plugin-prettier`
2. Run `npm install`
3. Run `npm run lint`

**Expected behavior**

`eslint` runs as expected.

**Desktop (please complete the following information):**
 - macOS 10.15.3
 - Terminal
 - node v8.17.0, npm 6.13.6.

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -33,6 +33,7 @@
 		"eslint-plugin-react": "^7.14.3",
 		"eslint-plugin-react-hooks": "^1.6.1",
 		"globals": "^12.0.0",
+		"prettier": "npm:wp-prettier@1.19.1",
 		"requireindex": "^1.2.0"
 	},
 	"peerDependencies": {

